File: frontend/src/types/experience.ts
export interface WorkExperience {
  id: number
  company: string
  position: string
  period: string
  description: string
  technologies: string[]
  achievements: string[]
  companyUrl?: string
  location: string
  current: boolean
}

export interface Education {
  id: number
  institution: string
  degree: string
  field: string
  period: string
  description: string
  location: string
}

File: backend/models/experience.go
package models

type WorkExperience struct {
	ID           int      `json:"id"`
	Company      string   `json:"company"`
	Position     string   `json:"position"`
	Period       string   `json:"period"`
	Description  string   `json:"description"`
	Technologies []string `json:"technologies"`
	Achievements []string `json:"achievements"`
	CompanyUrl   string   `json:"companyUrl,omitempty"`
	Location     string   `json:"location"`
	Current      bool     `json:"current"`
}

type Education struct {
	ID          int    `json:"id"`
	Institution string `json:"institution"`
	Degree      string `json:"degree"`
	Field       string `json:"field"`
	Period      string `json:"period"`
	Description string `json:"description"`
	Location    string `json:"location"`
}
